### PR TITLE
gh-753: Update travis to reflect release proc

### PR DESCRIPTION
**Pull request by bobvanluijt**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,32 +1,46 @@
...
-      travis_wait 30 docker push creativesoftwarefdn/weaviate:unstable
+    # only push to Dockerhub if master and not a PR
+    if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then
+        # Login to dockerhub
+        echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
+        # prepare release id's
+        DOCKER_ID="creativesoftwarefdn/weaviate:"
+        # releasing a prerelease
+        if [ "$TRAVIS_BRANCH" = "master" ]; then
+            RELEASE_ID=$DOCKER_ID$WEAVIATE_VERSION-$(echo $TRAVIS_COMMIT | cut -c1-5)
```
Comment (etiennedi) [inline .travis.yml]: How many digits is that? The git short hash should have 7 digits/characters.

Comment (etiennedi) [reply .travis.yml]: better name for `RELEASE_ID` would probably be `TAG` to stay consistent with Docker's domain language.

File: .travis.yml [hunk 2]
```diff
@@ -1,32 +1,46 @@
...
+            RELEASE_ID=$DOCKER_ID$TRAVIS_TAG
+            DOCKERTARGET=janus # <=== using JanusGraph by default now
+            echo "preparing release: " $RELEASE_ID " for target: " $DOCKERTARGET
+            # validate if the tag is equal to the Weaviate version
+            if [ "$TRAVIS_TAG" = "$WEAVIATE_VERSION" ]; then
+                # execute the Docker build
+                travis_wait 30 docker build -f "Dockerfile-prod" -t $RELEASE_ID --target $DOCKERTARGET .
+            # something is wrong with the tags!
+            else
+                echo "The release tag ($TRAVIS_TAG) and Weaviate version ($WEAVIATE_VERSION) are not equal! Can't release."
```
Comment (etiennedi) [inline .travis.yml]: Nice!

Comment (bobvanluijt) [reply .travis.yml]: Thanks :-)

File: .travis.yml (hunk 1 above)
Comment (bobvanluijt) [reply .travis.yml]: Check

**Pull request closed without merging**